OSDOCS#9533:Updates for preflight cluster upgrades for ROSA Classic. (#76632)
openshift/openshift-docs: upgrading/rosa-upgrading-sts.adoc

:_mod-docs-content-type: ASSEMBLY
[id="rosa-upgrading-sts"]
= Upgrading ROSA (classic architecture) clusters
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: rosa-upgrading-sts

toc::[]

Use one of the following methods to upgrade ROSA (classic architecture) clusters:

* Manually via the ROSA CLI (`rosa`) - Start a one-time immediate upgrade or schedule a one-time upgrade for a future date or time.
* Manually via the {cluster-manager} UI - Start a one-time immediate upgrade or schedule a one-time upgrade for a future date or time.

> * Automated upgrades - Set an upgrade window for recurring y-stream upgrades whenever a new version is available without needing to manually schedule it. Minor versions have to be manually scheduled.

Reviewer: ```suggestion
* Automated upgrades - Set an upgrade window for automatic recurring y-stream upgrades whenever a new version is available. Upgrades to minor versions must be manually scheduled.
```
Reworded for clarity

[id="rosa-lifecycle-policy_{context}"]
== Life cycle policies and planning

To plan an upgrade, review the xref:../rosa_architecture/rosa_policy_service_definition/rosa-life-cycle.adoc#rosa-life-cycle[{product-title} update life cycle]. The life cycle page includes release definitions, support and upgrade requirements, installation policy information and life cycle dates.

> OpenShift update channels are the mechanism that you can use to decide which Red Hat OpenShift Container Platform minor version to update your clusters to. {product-title} only supports updates through the `stable` channel. To learn more about OpenShift update channels and releases, see link:https://docs.openshift.com/container-platform/4.15/updating/understanding_updates/understanding-update-channels-release.html[Understanding update channels and releases].

Reviewer: ```suggestion
You may use update channels to decide which Red Hat OpenShift Container Platform minor version to update your clusters to. {product-title} only supports updates through the `stable` channel. To learn more about OpenShift update channels and releases, see link:https://docs.openshift.com/container-platform/4.15/updating/understanding_updates/understanding-update-channels-release.html[Understanding update channels and releases].
```

Red Hat hosts a public OpenShift Update Service (OSUS), which provides a link:https://access.redhat.com/labs/ocpupgradegraph/update_channel[graph] of update possibilities based on the {product-title}(ROSA) release images in the official registry.
[NOTE]
====
Certain upgrade paths available in Red Hat OpenShift Container Platform might not be available in ROSA.
====

include::modules/rosa-how-upgrades-work.adoc[leveloffset=+1]

[id="rosa-sts-upgrading-a-cluster-with-sts"]
== Upgrading a ROSA Classic cluster

You must update {product-title} (ROSA) Classic clusters using the ROSA CLI (`rosa`) or the {cluster-manager} console.

[NOTE]
====
The actual start time of the cluster upgrade will be within one hour of the upgrade schedule time. Additionally, the duration of the upgrade might vary based on your workload configuration.
====

> When a ROSA Classic cluster that uses AWS Security Token Services (STS) is upgraded, the ROSA CLI verifies the account and operator role policies for the chosen cluster are compatible with the target version of the upgrade. If the policies are compatible, the CLI automatically upgrades the IAM policies. If the policies are not compatible with the chosen upgrade version, the CLI prompts you to upgrade them. When scheduling the upgrade, you provide administrative acknowledgment to automatically upgrade these policies, if required.

Reviewer 2: the sentence "If the policies are compatible, the CLI automatically upgrades the cluster"

Reviewer 2: "When scheduling the upgrade, you provide administrative acknowledgment to confirm you have reviewed the changes involved with the upgrade, if required."
